src/MendelIHT_utilities.jl

```diff
@@ -168,4 +183,40 @@ function std_reciprocal(A::SnpArray, mean_vec::Vector{Float64})
     return std_vector
 end
 
+""" Projects the point y onto the set with at most m active groups and at most 
+n active predictors per group. The variable group encodes group membership. Currently
+assumes there are no unknown group membership."""
+function doubly_sparse_projection(y::Vector{Float64}, group::Vector{Int64}, m::Int64, n::Int64)
+    x = copy(y)
+    groups = maximum(group)
+    group_count = zeros(Int, groups)         #counts number of predictors in each group
+    z = zeros(groups)                        #l2 norm of each group
+    perm = sortperm(x, by = abs, rev = true) 
+
+    #calculate the magnitude of each group, where only top predictors contribute
+    for i = 1:length(x)
+        j = perm[i] 
+        k = group[j]
+        if group_count[k] < n
+            z[k] = z[k] + x[j]^2
+            group_count[k] = group_count[k] + 1
+        end
+    end
+
+    #go through the top predictors in order. Set predictor to 0 if criteria not met
```

Comment:
if you only need to go through top predictors, then `sortperm` is inefficient because it sorts the entire vector. something like `selectperm!` may be more efficient

Reply:
Ok, but this might be a bit tricky because I'm not sure when I should stop sorting, maybe stop at J*k? I'll think about how to do it